src/coreclr/inc/pgo_formatprocessing.h

```diff
@@ -383,13 +382,17 @@ inline bool CheckIfPgoSchemaIsCompatibleAndSetOffsets(const uint8_t *pByte, size
             schemaTable[nMatched].Offset = schema.Offset;
             nMatched++;
         }
+        else
+        {
+            return false;
```

Comment:
nit: return true/false looks a bit off for `size_t` return type

Reply:
The return true/false is for the "format processing functor", not for the method as a whole. The false is saying we can bail out of walking the schema when we find the first mismatch.